ci/build_python.sh

```diff
@@ -4,8 +4,11 @@ set -e -E -u -o pipefail
 
 rapids-print-env
 
+rapids-generate-version > ./VERSION
```

Comment:
This is actually not related to the version of RAPIDS (e.g. the version used to pull in `rapids-cmake`), though I understand how the command names make it seem that way 😅 

`rapids-generate-version` comes from here: https://github.com/rapidsai/gha-tools/blob/8bd8fca71b5fae38b1493c547d15e73da40b32e1/tools/rapids-generate-version

It's just a small shell script that computes a version string from git tags, like this:

```shell
dunamai_format="{base}"
if ! rapids-is-release-build; then
    # Nightlies include the distance from the last tag as the alpha version.
    dunamai_format="{base}{stage}{distance}"
fi

# Now change the version.
dunamai_version=$(python -m dunamai from git --format ${dunamai_format})

echo -n "${dunamai_version}"
```

I'm proposing using that tool to avoid duplicating that logic here in `legate-boost`.

The `rapids-*` prefix on all the tools from that repo is primarily intended to avoid conflicts with other tools (`generate-version` is a pretty generic name, for example)... it doesn't by itself mean that a particular tool does something RAPIDS-specific.